src/Control/Email/Email.php

```diff
@@ -706,30 +719,39 @@ public function setBody($body)
     }
 
     /**
+     * @deprecated 4.12.0 Will be replaced with Symfony\Component\Mime\Email::html()
+     *
      * @return $this
      */
     public function invalidateBody()
     {
+        Deprecation::notice('4.12.0', 'Will be replaced with Symfony\Component\Mime\Email::html()');
         $this->setBody(null);
 
         return $this;
     }
 
     /**
+     * @deprecated 4.12.0 Will be added to return value of getData()
+     *
      * @return string The base URL for the email
      */
     public function BaseURL()
     {
+        Deprecation::notice('4.12.0', 'Will be added to return value of getData()');
```

Comment:
```suggestion
        Deprecation::notice('4.12.0', 'Use getData() instead');
```

Reply:
Why does `getData()` return the absolute base URL? And what's wrong with keeping this method implementation?

Reply:
In `4`, Email extends ViewableData and the Email instance is rendered directly in a template.

In `5`, things have changed a lot and Email extends `Symfony\Components\Mime\Email`.  When rendering to templates, a ViewableData instance is temporarily created and the results of getData() are passed to the instance.

Hence why we no longer need the `BaseURL()` and `IsEmail()` methods here - they're added to getData() here https://github.com/silverstripe/silverstripe-framework/pull/10494/files#diff-6c9d9301adeda8e2aa1c06db7e907f0d57ee1ba23fc43ff685eef9d225d0a9d2R303

Reply:
Fair enough. We should still use the new standard "Will be replaced with getData() in 5.0.0" syntax here.

Reply:
Updated